src/components/global/DialogForm.vue

```diff
@@ -0,0 +1,173 @@
+ */
+
+// libraries
+import { defineComponent, computed } from 'vue';
+import { QForm } from 'quasar';
+
+export default defineComponent({
+  name: 'DialogForm',
+  props: {
+    formRef: {
+      type: Object as () => typeof QForm | null,
+      required: false,
+    },
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  emits: ['change', 'form-submit', 'update:modelValue'],
+  setup(props, { emit }) {
+    // determines if dialog window is open
+    const isOpen = computed({
+      get: (): boolean => props.modelValue,
+      set: (value: boolean): void => {
+        emit('update:modelValue', value);
+        emit('change');
+      },
+    });
+
+    // closes the dialog window
+    const close = (): void => {
+      if (props.formRef) {
+        props.formRef.reset();
+      }
+      isOpen.value = false;
+    };
+
+    /**
+     * Validates the form.
+     * If form is valid emits `form-submit` event and closes dialog.
+     */
+    const submit = async (): Promise<void> => {
+      if (props.formRef) {
+        const isFormValid: boolean = await props.formRef.validate();
+
+        if (isFormValid) {
+          emit('form-submit');
+          close();
+        } else {
+          props.formRef.$el.scrollIntoView({
+            behavior: 'smooth',
+          });
+        }
+      }
+    };
+
+    return {
+      isOpen,
+      close,
+      submit,
+    };
+  },
+});
+</script>
+
+<template>
+  <q-dialog square persistent v-model="isOpen" data-cy="dialog-form">
+    <q-card
+      class="relative-position overflow-visible bg-white"
+      style="min-width: 50vw"
+    >
+      <!-- Section: Card header -->
+      <q-card-section
+        v-if="$slots.title"
+        class="q-pt-none"
+        data-cy="dialog-header"
+      >
+        <!-- Title -->
+        <h3 class="text-h6 q-mt-sm q-pt-xs q-mb-none">
+          <slot name="title" />
+        </h3>
+      </q-card-section>
+
+      <q-separator />
+
+      <!-- Section: Card body -->
+      <q-card-section
+        v-if="$slots.content || $slots.buttons"
+        class="scroll"
+        data-cy="dialog-body"
+        style="max-height: 80vh"
+      >
+        <!-- Content -->
+        <slot v-if="$slots.content" name="content"></slot>
+        <!-- Buttons -->
+        <div v-if="$slots.buttons" class="flex justify-end q-mt-sm">
+          <slot name="buttons" :close="close" :submit="submit"></slot>
+        </div>
+      </q-card-section>
+
+      <!-- Button: Close dialog -->
+      <q-card-actions
+        class="dialog-close__wrapper inline-block absolute-top-right q-px-none q-py-none"
+      >
+        <q-btn
+          v-close-popup
+          round
+          unelevated
+          color="blue-grey-1"
+          icon="close"
+          class="dialog-close text-grey-10"
+          data-cy="dialog-close"
+        />
+      </q-card-actions>
+    </q-card>
+  </q-dialog>
+</template>
+
+<style lang="scss" scoped>
+.q-dialog__inner > div {
```

Comment:
Style is duplicated with `DialogCard` component. https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/270efb98d2830ab25af92092282b63e86280fbdc/src/components/global/DialogCard.vue#L132-L148.

Reply:
Reworked into a `DialogDefault` component.